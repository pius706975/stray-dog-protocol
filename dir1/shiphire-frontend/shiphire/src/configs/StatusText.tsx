export const StatusText = {
    'rfq 1': 'Waiting for ship owner to respond',
    'rfq 2': 'Waiting on your proposal',
    'proposal 1': 'Waiting for your agreement',
    'proposal 2': 'Waiting for your payment',
    'proposal 3': 'Waiting for payment approval',
    'proposal 4': 'Waiting for contract',
    'contract 1': 'Waiting for your signature',
};
