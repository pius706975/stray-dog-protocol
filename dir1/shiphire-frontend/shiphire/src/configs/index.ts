export * from "./Fonts"
export * from "./Colors"
export * from "./LocalStorage"
export * from "./Svg"
export * from "./StatusText"
export * from "./firebase/GoogleSignIn"
export * from "./firebase/AppleSignIn"