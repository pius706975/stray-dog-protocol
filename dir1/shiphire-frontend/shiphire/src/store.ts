import { configureStore } from '@reduxjs/toolkit';
import {
    addShipSlice,
    documentNameSlice,
    modalSlice,
    networkErrorSlice,
    notifBadgeSlice,
    progressIndicatorSlice,
    renterPreferenceSlice,
    requestForQuoteSlice,
    shipSpecSlice,
    termsStatusSlice,
    userLanguageSlice,
    userLocationSlice,
    userStatusSlice,
} from './slices';

const store = configureStore({
    reducer: {
        modal: modalSlice.reducer,
        progressIndicator: progressIndicatorSlice.reducer,
        renterPreference: renterPreferenceSlice.reducer,
        requestForQuote: requestForQuoteSlice.reducer,
        userStatus: userStatusSlice.reducer,
        addShip: addShipSlice.reducer,
        documentName: documentNameSlice.reducer,
        termsStatus: termsStatusSlice.reducer,
        userLanguage: userLanguageSlice.reducer,
        userLocation: userLocationSlice.reducer,
        shipSpecSlice: shipSpecSlice.reducer,
        notifBadgeSlice: notifBadgeSlice.reducer,
        networkError: networkErrorSlice.reducer,
    },
});

export default store;
