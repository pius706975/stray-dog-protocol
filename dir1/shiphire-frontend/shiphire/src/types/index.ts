export * from './Components'
export * from './Navigators'
export * from './ReduxToolkit'
export * from './Screens'
export * from './Services'