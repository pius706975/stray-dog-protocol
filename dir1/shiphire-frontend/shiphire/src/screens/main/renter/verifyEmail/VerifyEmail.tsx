import React from 'react';
import { Pressable, Text, TextInput } from 'react-native';
import { View } from 'react-native-ui-lib';
import { useDispatch } from 'react-redux';
import { Button, CustomText, ScreenLayout } from '../../../../components';
import {
    Color,
    FontFamily,
    FontSize,
    OTPIcon,
    USERDATA,
    getDataFromLocalStorage,
    setDataToLocalStorage,
} from '../../../../configs';
import { useSendOTPEmailVerif, useVerifyEmailOTP } from '../../../../hooks';
import { modalSlice } from '../../../../slices';
import { VerifyEmailProps } from '../../../../types';
import { handleAxiosError } from '../../../../utils';
import { useTranslation } from 'react-i18next';

const VerifyEmail: React.FC<VerifyEmailProps> = ({ navigation }) => {
    const dispatch = useDispatch();
    const { t } = useTranslation('verifyemail');
    const { hideModal, showModal } = modalSlice.actions;
    const mutationSendOTPEmailVerif = useSendOTPEmailVerif();
    const mutationVerifEmailOTP = useVerifyEmailOTP();
    const [pinReady, setPinReady] = React.useState<boolean>(false);
    const [countdown, setCountdown] = React.useState<number>(0);
    const [userEmail, setUserEmail] = React.useState<string>('');
    const [countdownActive, setCountdownActive] =
        React.useState<boolean>(false);
    const [code, setCode] = React.useState<string>('');
    const [inputContainerIsFocused, setInputContainerIsFocused] =
        React.useState<boolean>(false);
    const textInputRef = React.useRef<any>(null);
    const codeDigitsArray = new Array(4).fill(0);

    const handleOnPress = () => {
        setInputContainerIsFocused(true);
        textInputRef?.current?.focus();
    };

    const handleOnBlur = () => {
        setInputContainerIsFocused(false);
    };

    const startCountdown = () => {
        setCountdown(60);
        setCountdownActive(true);
        mutationSendOTPEmailVerif.mutate(undefined, {
            onSuccess: () => {
                // hideProgressIndicator();

                dispatch(
                    showModal({
                        status: 'info',
                        text: t('infoNewCodeSent'),
                    }),
                );
                setTimeout(() => {
                    dispatch(hideModal());
                }, 4000);
            },
            onError: err => {
                // hideProgressIndicator();
                dispatch(
                    showModal({
                        status: 'failed',
                        text: t('AccountHeader.failedSendEmailVerif'),
                    }),
                );
                handleAxiosError(err);
            },
        });
    };

    const handleVerifySent = () => {
        let errorText = 'Request Failed';
        mutationVerifEmailOTP.mutate(
            { emailVerifOTP: +code },
            {
                onSuccess: resp => {
                    dispatch(
                        showModal({
                            status: 'success',
                            text: t('successVerifyEmail'),
                        }),
                    );
                    getDataFromLocalStorage(USERDATA).then(resp => {
                        console.log('resp', resp);
                        setDataToLocalStorage(USERDATA, {
                            ...resp,
                            isVerified: true,
                        });
                        console.log('isVerifiedEmail', resp.isVerified);
                    });
                    setTimeout(() => {
                        dispatch(hideModal());
                        navigation.navigate('Account');
                    }, 4000);
                },
                onError: err => {
                    handleAxiosError(err);
                    if (err.response?.status === 422) {
                        errorText = t('failedInvalidOTP');
                    }
                    dispatch(showModal({ status: 'failed', text: errorText }));
                    setTimeout(() => {
                        dispatch(hideModal());
                    }, 4000);
                    setCode('');
                },
            },
        );
    };

    const toCodeDigitInput = (_value: number, index: number) => {
        const emptyInputChar = ' ';
        const digit = code[index] || emptyInputChar;

        const isCurrentDigit = index === code.length;
        const isLastDigit = index === 3;
        const isCodeFull = code.length === 3;

        const isDigitFocused = isCurrentDigit || (isLastDigit && isCodeFull);

        return (
            <View
                key={index}
                style={{
                    borderColor:
                        inputContainerIsFocused && isDigitFocused
                            ? Color.secColor
                            : Color.primaryColor,
                    backgroundColor:
                        inputContainerIsFocused && isDigitFocused
                            ? Color.softSecColor
                            : Color.bgColor,
                    minWidth: '15%',
                    borderWidth: 3,
                    borderRadius: 5,
                    padding: 12,
                }}>
                <CustomText
                    fontSize="md"
                    fontFamily="semiBold"
                    color="darkTextColor"
                    textAlign="center">
                    {digit}
                </CustomText>
            </View>
        );
    };

    React.useEffect(() => {
        let countdownInterval: any;
        if (countdownActive) {
            countdownInterval = setInterval(() => {
                setCountdown(prevCountdown => prevCountdown - 1);
            }, 1000);
        }

        if (countdown === 0 || !countdownActive) {
            clearInterval(countdownInterval);
            setCountdownActive(false);
        }

        return () => {
            clearInterval(countdownInterval);
        };
    }, [countdown, countdownActive]);

    React.useEffect(() => {
        setPinReady(code.length === 4);
        return () => setPinReady(false);
    }, [code]);

    React.useEffect(() => {
        getDataFromLocalStorage(USERDATA).then(resp => {
            setUserEmail(resp.email);
        });
        dispatch(
            showModal({
                status: 'info',
                text: t('infoSentOTPCode'),
            }),
        );
        setTimeout(() => {
            dispatch(hideModal());
        }, 5000);
    }, []);

    return (
        <ScreenLayout
            flex
            center
            backgroundColor="light"
            testId="VerifyEmailScreen">
            <View
                center
                style={{
                    marginVertical: 30,
                }}>
                <OTPIcon />
                <CustomText
                    fontSize="xl"
                    fontFamily="bold"
                    color="darkTextColor">
                    {t('textEmailVerification')}
                </CustomText>
                <View paddingH-20>
                    <CustomText
                        fontSize="sm"
                        fontFamily="regular"
                        color="darkTextColor"
                        textAlign="center">
                        {t('textEnterOTP')}{' '}
                        <Text style={{ fontFamily: FontFamily.bold }}>
                            {userEmail}
                        </Text>{' '}
                        {t('textToVerify')}
                    </CustomText>
                </View>
                <View
                    center
                    style={{
                        marginVertical: 30,
                    }}>
                    <Pressable
                        testID='pressCode'
                        style={{
                            width: '70%',
                            flexDirection: 'row',
                            justifyContent: 'space-around',
                        }}
                        onPress={handleOnPress}>
                        {codeDigitsArray.map(toCodeDigitInput)}
                    </Pressable>
                    <TextInput
                        testID='codeInput'
                        style={{
                            position: 'absolute',
                            width: '70%',
                            height: 10,
                            opacity: 0,
                        }}
                        value={code}
                        onChangeText={text => setCode(text)}
                        keyboardType="number-pad"
                        maxLength={4}
                        returnKeyType="done"
                        textContentType="oneTimeCode"
                        onBlur={handleOnBlur}
                    />
                </View>
                <CustomText
                    fontSize="sm"
                    fontFamily="semiBold"
                    color="darkTextColor">
                    {t('textDidnt')}
                </CustomText>
                {countdown === 0 ? (
                    <Pressable onPress={startCountdown} testID='btnCount'>
                        <CustomText
                            fontSize="sm"
                            fontFamily="semiBold"
                            color="primaryColor"
                            borderBottomColor={Color.primaryColor}>
                            {t('textResend')}
                        </CustomText>
                    </Pressable>
                ) : (
                    <Text
                        style={{
                            fontFamily: FontFamily.regular,
                            fontSize: FontSize.sm,
                            color: Color.darkTextColor,
                        }}>
                        {t('textResendAgain')}{' '}
                        <Text
                            style={{
                                fontFamily: FontFamily.semiBold,
                                color: Color.primaryColor,
                            }}>
                            {countdown}s
                        </Text>
                    </Text>
                )}
                <View
                    row
                    style={{
                        marginTop: 20,
                        paddingHorizontal: 30,
                    }}>
                    <View flex>
                        <Button
                            testID='buttonVerify'
                            title={t('labelButtonVerify')}
                            disable={!pinReady}
                            onSubmit={handleVerifySent}
                        />
                    </View>
                </View>
            </View>
        </ScreenLayout>
    );
};

export default VerifyEmail;
