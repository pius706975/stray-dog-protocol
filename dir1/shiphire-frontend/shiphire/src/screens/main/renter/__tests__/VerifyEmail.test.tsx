import { createNativeStackNavigator } from "@react-navigation/native-stack";
import { QueryClient, QueryClientProvider } from "react-query";
import { MainStackParamList } from "../../../../types";
import MockAdapter from "axios-mock-adapter";
import httpRequest from "../../../../services/api";
import { Provider } from "react-redux";
import store from "../../../../store";
import { NavigationContainer } from "@react-navigation/native";
import { VerifyEmail } from "../verifyEmail";
import AsyncStorage from "@react-native-async-storage/async-storage";
import { TOKEN, USERDATA } from "../../../../configs";
import { fireEvent, render, waitFor } from "@testing-library/react-native";
import { act } from "react-test-renderer";

const queryClient = new QueryClient();
const MainStack = createNativeStackNavigator<MainStackParamList>();
const mockAdapter = new MockAdapter(httpRequest);

const verifStackNavMockComponent = () => (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainStack.Navigator>
                    <MainStack.Screen name="VerifEmail" component={VerifyEmail} />
                </MainStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);

AsyncStorage.setItem(
    USERDATA,
    JSON.stringify({
        name: 'Muhamad Fauzan',
        email: 'usman@email.com',
        password: 'test',
        isVerified: false,
    }),
);
AsyncStorage.setItem(
    TOKEN,
    JSON.stringify({
        accessToken: 'test',
        refreshToken: 'test',
    }),
);

describe('Verif email screen', () => {
    it('should renders correctly',async () => {
        const tree = render(verifStackNavMockComponent()).toJSON();
        await act(async() => {
            expect(tree).toMatchSnapshot();
        })
    });
});

describe('Verif email component', () => {
    it('should render the screen',async () => {
        const {getByTestId} = render(verifStackNavMockComponent());

        await act(async() => {
            expect(getByTestId('VerifyEmailScreen')).toBeDefined();
        })
    })
    it('should pressCode on the screen', async () => {
        const {getByTestId} = render(verifStackNavMockComponent());

        const pressCode = getByTestId('pressCode')

        await act(async() => {
            fireEvent.press(pressCode)
        })

        const code = getByTestId('codeInput')

        expect(code).toBeDefined();
        await act(async () => {
            fireEvent.changeText(code, '1234')
        }) 
    })
    it('should show hide the modal after render API', async () => {
        jest.useFakeTimers();
        const {getByTestId} = render(verifStackNavMockComponent());

        const pressCode = getByTestId('pressCode')

        await act(async() => {
            fireEvent.press(pressCode)
        })

        const code = getByTestId('codeInput')

        expect(code).toBeDefined();
        await act(async () => {
            fireEvent.changeText(code, '1234')
        }) 

        await waitFor(async () => {
            const modalState = store.getState().modal;
            jest.advanceTimersByTime(4000);
            expect(modalState).toEqual({
                visible: false,
                status: undefined,
                text: ''
            });
        });
    })
    it('should verify the code', async () => {
        jest.useFakeTimers();
        const {getByTestId} = render(verifStackNavMockComponent());

        const pressCode = getByTestId('pressCode')

        await act(async () => {
            fireEvent.press(pressCode)
        })

        const code = getByTestId('codeInput')

        expect(code).toBeDefined();
        await act(async () => {
            fireEvent.changeText(code, '1234')
        }) 

        fireEvent.press(getByTestId('buttonVerify'))

        mockAdapter
            .onPost('/renter/verify-email-otp')
            .reply(200, { status: 'success' })

            await waitFor(async () => {
                const modalState = store.getState().modal;
                jest.advanceTimersByTime(4000);
                expect(modalState).toEqual({
                    visible: true,
                    status: 'success',
                    text: 'successVerifyEmail'
                });
            });
    })
    it('should failed verify the code', async () => {
        jest.useFakeTimers();
        const {getByTestId} = render(verifStackNavMockComponent());

        await act(async () => {
            fireEvent.press(getByTestId('buttonVerify'))
        })

        mockAdapter
            .onPost('/renter/verify-email-otp')
            .reply(500, { status: 'failed' })
    })
    it('should resend code', async () => {
        jest.useFakeTimers();
        const {getByTestId} = render(verifStackNavMockComponent());

        await act(async() => {
            fireEvent.press(getByTestId('btnCount'))
        })

        mockAdapter
            .onGet('/renter/send-otp-email-verif')
            .reply(200, {status: 'success'})
    })
})