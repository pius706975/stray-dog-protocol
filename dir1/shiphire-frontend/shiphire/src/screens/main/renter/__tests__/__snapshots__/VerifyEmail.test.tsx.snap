// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verif email screen should renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                undefined,
                undefined,
                {
                  "flex": 1,
                },
                undefined,
                {},
                {},
                {},
                {
                  "backgroundColor": "#f0f6fb",
                  "justifyContent": "space-between",
                  "position": "relative",
                },
              ]
            }
            testID="VerifyEmailScreen"
          >
            <RCTScrollView
              contentContainerStyle={
                {
                  "alignItems": "center",
                  "flex": 1,
                  "flexGrow": 1,
                  "gap": undefined,
                  "justifyContent": "center",
                  "marginBottom": undefined,
                  "padding": undefined,
                  "paddingVertical": undefined,
                }
              }
            >
              <View>
                <View
                  center={true}
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {},
                      {},
                      {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                      {
                        "marginVertical": 30,
                      },
                    ]
                  }
                >
                  <RNSVGSvgView
                    align="xMidYMid"
                    bbHeight="140"
                    bbWidth="107"
                    fill="none"
                    focusable={false}
                    height={140}
                    meetOrSlice={0}
                    minX={0}
                    minY={0}
                    style={
                      [
                        {
                          "backgroundColor": "transparent",
                          "borderWidth": 0,
                        },
                        {
                          "flex": 0,
                          "height": 140,
                          "width": 107,
                        },
                      ]
                    }
                    vbHeight={140}
                    vbWidth={107}
                    width={107}
                    xml="<svg width="107" height="140" viewBox="0 0 107 140" fill="none" xmlns="http://www.w3.org/2000/svg">
    <path d="M53.0315 42.4219L92.3808 65.9678C93.5829 66.6042 94.396 67.4704 94.8203 68.5664C95.2445 69.6624 95.4567 70.7407 95.4567 71.8013V120.908C95.4567 122.605 94.8203 124.09 93.5475 125.363C92.2748 126.636 90.7899 127.272 89.0929 127.272H16.9702C15.2732 127.272 13.7883 126.636 12.5155 125.363C11.2428 124.09 10.6064 122.605 10.6064 120.908V71.8013C10.6064 70.7407 10.8362 69.6624 11.2958 68.5664C11.7554 67.4704 12.5509 66.6042 13.6822 65.9678L53.0315 42.4219ZM53.0315 91.8472L88.6686 70.9528L53.0315 49.5281L17.3944 70.9528L53.0315 91.8472ZM53.0315 98.9534L16.9702 77.7408V120.908H89.0929V77.7408L53.0315 98.9534ZM53.0315 120.908H89.0929H16.9702H53.0315Z" fill="#4D5963" fill-opacity="0.588235"/>
    <rect width="106.063" height="35.3543" rx="5" fill="#216DAB"/>
    <path d="M36.9131 28.9699V26.3231H71.6246V28.9699H36.9131ZM38.5619 17.949L36.8263 16.9944L38.5619 14.044H35.1776V12.0914H38.5619L36.8263 9.14095L38.5619 8.18639L40.2107 11.0935L41.8595 8.18639L43.5951 9.14095L41.8595 12.0914H45.2439V14.044H41.8595L43.5951 16.9944L41.8595 17.949L40.2107 15.0419L38.5619 17.949ZM52.6201 17.949L50.8845 16.9076L52.6201 13.9572H49.2357V12.0047H52.6201L50.8845 9.05418L52.6201 8.09961L54.2689 11.0067L55.9177 8.09961L57.6532 9.05418L55.9177 12.0047H59.302V13.9572H55.9177L57.6532 16.9076L55.9177 17.949L54.2689 15.0419L52.6201 17.949ZM66.6782 17.949L64.9426 16.9076L66.6782 13.9572H63.2939V12.0047H66.6782L64.9426 9.05418L66.6782 8.09961L68.327 11.0067L69.9758 8.09961L71.7114 9.05418L69.9758 12.0047H73.3602V13.9572H69.9758L71.7114 16.9076L69.9758 17.949L68.327 15.0419L66.6782 17.949Z" fill="#F0F6FB"/>
    </svg>        
    "
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <RNSVGGroup
                      fill={null}
                      propList={
                        [
                          "fill",
                        ]
                      }
                    >
                      <RNSVGPath
                        d="M53.0315 42.4219L92.3808 65.9678C93.5829 66.6042 94.396 67.4704 94.8203 68.5664C95.2445 69.6624 95.4567 70.7407 95.4567 71.8013V120.908C95.4567 122.605 94.8203 124.09 93.5475 125.363C92.2748 126.636 90.7899 127.272 89.0929 127.272H16.9702C15.2732 127.272 13.7883 126.636 12.5155 125.363C11.2428 124.09 10.6064 122.605 10.6064 120.908V71.8013C10.6064 70.7407 10.8362 69.6624 11.2958 68.5664C11.7554 67.4704 12.5509 66.6042 13.6822 65.9678L53.0315 42.4219ZM53.0315 91.8472L88.6686 70.9528L53.0315 49.5281L17.3944 70.9528L53.0315 91.8472ZM53.0315 98.9534L16.9702 77.7408V120.908H89.0929V77.7408L53.0315 98.9534ZM53.0315 120.908H89.0929H16.9702H53.0315Z"
                        fill={
                          {
                            "payload": 4283259235,
                            "type": 0,
                          }
                        }
                        fillOpacity={0.588235}
                        propList={
                          [
                            "fill",
                            "fillOpacity",
                          ]
                        }
                      />
                      <RNSVGRect
                        fill={
                          {
                            "payload": 4280380843,
                            "type": 0,
                          }
                        }
                        height="35.3543"
                        propList={
                          [
                            "fill",
                          ]
                        }
                        rx="5"
                        width="106.063"
                        x="0"
                        y="0"
                      />
                      <RNSVGPath
                        d="M36.9131 28.9699V26.3231H71.6246V28.9699H36.9131ZM38.5619 17.949L36.8263 16.9944L38.5619 14.044H35.1776V12.0914H38.5619L36.8263 9.14095L38.5619 8.18639L40.2107 11.0935L41.8595 8.18639L43.5951 9.14095L41.8595 12.0914H45.2439V14.044H41.8595L43.5951 16.9944L41.8595 17.949L40.2107 15.0419L38.5619 17.949ZM52.6201 17.949L50.8845 16.9076L52.6201 13.9572H49.2357V12.0047H52.6201L50.8845 9.05418L52.6201 8.09961L54.2689 11.0067L55.9177 8.09961L57.6532 9.05418L55.9177 12.0047H59.302V13.9572H55.9177L57.6532 16.9076L55.9177 17.949L54.2689 15.0419L52.6201 17.949ZM66.6782 17.949L64.9426 16.9076L66.6782 13.9572H63.2939V12.0047H66.6782L64.9426 9.05418L66.6782 8.09961L68.327 11.0067L69.9758 8.09961L71.7114 9.05418L69.9758 12.0047H73.3602V13.9572H69.9758L71.7114 16.9076L69.9758 17.949L68.327 15.0419L66.6782 17.949Z"
                        fill={
                          {
                            "payload": 4293981947,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                      />
                    </RNSVGGroup>
                  </RNSVGSvgView>
                  <Text
                    numberOfLines={0}
                    style={
                      {
                        "borderBottomWidth": 0,
                        "borderColor": undefined,
                        "color": "#4d5963",
                        "fontFamily": "Kanit-Bold",
                        "fontSize": 22,
                        "lineHeight": undefined,
                        "textAlign": undefined,
                      }
                    }
                  >
                    textEmailVerification
                  </Text>
                  <View
                    paddingH-20={true}
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {
                          "paddingHorizontal": 20,
                        },
                        {},
                        {},
                        undefined,
                      ]
                    }
                  >
                    <Text
                      numberOfLines={0}
                      style={
                        {
                          "borderBottomWidth": 0,
                          "borderColor": undefined,
                          "color": "#4d5963",
                          "fontFamily": "Kanit-Regular",
                          "fontSize": 16,
                          "lineHeight": undefined,
                          "textAlign": "center",
                        }
                      }
                    >
                      textEnterOTP
                       
                      <Text
                        style={
                          {
                            "fontFamily": "Kanit-Bold",
                          }
                        }
                      />
                       
                      textToVerify
                    </Text>
                  </View>
                  <View
                    center={true}
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {},
                        {},
                        {
                          "alignItems": "center",
                          "justifyContent": "center",
                        },
                        {
                          "marginVertical": 30,
                        },
                      ]
                    }
                  >
                    <View
                      accessibilityState={
                        {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        }
                      }
                      accessibilityValue={
                        {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        }
                      }
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        {
                          "flexDirection": "row",
                          "justifyContent": "space-around",
                          "width": "70%",
                        }
                      }
                      testID="pressCode"
                    >
                      <View
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {
                              "backgroundColor": "#f0f6fb",
                              "borderColor": "#216dab",
                              "borderRadius": 5,
                              "borderWidth": 3,
                              "minWidth": "15%",
                              "padding": 12,
                            },
                          ]
                        }
                      >
                        <Text
                          numberOfLines={0}
                          style={
                            {
                              "borderBottomWidth": 0,
                              "borderColor": undefined,
                              "color": "#4d5963",
                              "fontFamily": "Kanit-SemiBold",
                              "fontSize": 18,
                              "lineHeight": undefined,
                              "textAlign": "center",
                            }
                          }
                        >
                           
                        </Text>
                      </View>
                      <View
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {
                              "backgroundColor": "#f0f6fb",
                              "borderColor": "#216dab",
                              "borderRadius": 5,
                              "borderWidth": 3,
                              "minWidth": "15%",
                              "padding": 12,
                            },
                          ]
                        }
                      >
                        <Text
                          numberOfLines={0}
                          style={
                            {
                              "borderBottomWidth": 0,
                              "borderColor": undefined,
                              "color": "#4d5963",
                              "fontFamily": "Kanit-SemiBold",
                              "fontSize": 18,
                              "lineHeight": undefined,
                              "textAlign": "center",
                            }
                          }
                        >
                           
                        </Text>
                      </View>
                      <View
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {
                              "backgroundColor": "#f0f6fb",
                              "borderColor": "#216dab",
                              "borderRadius": 5,
                              "borderWidth": 3,
                              "minWidth": "15%",
                              "padding": 12,
                            },
                          ]
                        }
                      >
                        <Text
                          numberOfLines={0}
                          style={
                            {
                              "borderBottomWidth": 0,
                              "borderColor": undefined,
                              "color": "#4d5963",
                              "fontFamily": "Kanit-SemiBold",
                              "fontSize": 18,
                              "lineHeight": undefined,
                              "textAlign": "center",
                            }
                          }
                        >
                           
                        </Text>
                      </View>
                      <View
                        style={
                          [
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            {
                              "backgroundColor": "#f0f6fb",
                              "borderColor": "#216dab",
                              "borderRadius": 5,
                              "borderWidth": 3,
                              "minWidth": "15%",
                              "padding": 12,
                            },
                          ]
                        }
                      >
                        <Text
                          numberOfLines={0}
                          style={
                            {
                              "borderBottomWidth": 0,
                              "borderColor": undefined,
                              "color": "#4d5963",
                              "fontFamily": "Kanit-SemiBold",
                              "fontSize": 18,
                              "lineHeight": undefined,
                              "textAlign": "center",
                            }
                          }
                        >
                           
                        </Text>
                      </View>
                    </View>
                    <TextInput
                      keyboardType="number-pad"
                      maxLength={4}
                      onBlur={[Function]}
                      onChangeText={[Function]}
                      returnKeyType="done"
                      style={
                        {
                          "height": 10,
                          "opacity": 0,
                          "position": "absolute",
                          "width": "70%",
                        }
                      }
                      testID="codeInput"
                      textContentType="oneTimeCode"
                      value=""
                    />
                  </View>
                  <Text
                    numberOfLines={0}
                    style={
                      {
                        "borderBottomWidth": 0,
                        "borderColor": undefined,
                        "color": "#4d5963",
                        "fontFamily": "Kanit-SemiBold",
                        "fontSize": 16,
                        "lineHeight": undefined,
                        "textAlign": undefined,
                      }
                    }
                  >
                    textDidnt
                  </Text>
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    testID="btnCount"
                  >
                    <Text
                      numberOfLines={0}
                      style={
                        {
                          "borderBottomWidth": 1,
                          "borderColor": "#216dab",
                          "color": "#216dab",
                          "fontFamily": "Kanit-SemiBold",
                          "fontSize": 16,
                          "lineHeight": undefined,
                          "textAlign": undefined,
                        }
                      }
                    >
                      textResend
                    </Text>
                  </View>
                  <View
                    row={true}
                    style={
                      [
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        {},
                        {},
                        {
                          "flexDirection": "row",
                        },
                        {
                          "marginTop": 20,
                          "paddingHorizontal": 30,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          undefined,
                          undefined,
                          {
                            "flex": 1,
                          },
                          undefined,
                          {},
                          {},
                          {},
                          undefined,
                        ]
                      }
                    >
                      <View
                        accessibilityState={
                          {
                            "busy": undefined,
                            "checked": undefined,
                            "disabled": true,
                            "expanded": undefined,
                            "selected": undefined,
                          }
                        }
                        accessibilityValue={
                          {
                            "max": undefined,
                            "min": undefined,
                            "now": undefined,
                            "text": undefined,
                          }
                        }
                        accessible={true}
                        collapsable={false}
                        focusable={true}
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          {
                            "alignItems": "center",
                            "backgroundColor": "#6499c4",
                            "borderRadius": 6,
                            "elevation": 2,
                            "flexDirection": "row",
                            "gap": 0,
                            "justifyContent": "center",
                            "paddingHorizontal": 10,
                            "paddingVertical": 6,
                          }
                        }
                        testID="buttonVerify"
                      >
                        <Text
                          numberOfLines={0}
                          style={
                            {
                              "borderBottomWidth": 0,
                              "borderColor": undefined,
                              "color": "#f0f6fb",
                              "fontFamily": "Kanit-Medium",
                              "fontSize": 18,
                              "lineHeight": undefined,
                              "textAlign": undefined,
                            }
                          }
                        >
                          labelButtonVerify
                        </Text>
                        <View
                          style={
                            [
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              undefined,
                              {
                                "position": "absolute",
                                "right": 20,
                              },
                            ]
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="VerifEmail"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
